003-OFEP-CUE-upstream.md

```diff
@@ -0,0 +1,22 @@
+## 003-OFEP-CUE-upstream
+
+## State: DRAFTING
+
+As flagD evolves, it will be presented on different types of services with a variety of protocols.
+Many of these protocols have client libraries that can be generated for convenience.
+For example, we use openapi to enable us to create a golang server implementation.
+This is great because a) it's fast and easy b) it matches the specification files promises exactly - validation/matching for free.
+
+Given we are looking to now expand to offer new generated server support for gRPC, we are in a predicament about whether to a) derive the implementation from OpenAPI b) have a new source of truth as .proto files c) have some sort of composite between the two.
+
+My proposal here would be to choose a new top-level DSL to define the specifications required to drive generated server code implementations. As such, cue offers integration with YAML/JSON & protobuf directly.
+This would mean we could use a single file(s) and toolchain to generate all of the automatically created code for flagD.
```

Comment:
nitpick: I'd like to get consensus on the casing style here. I have leaned toward "flagd" in the pattern of "etcd", "systemd", etc, but consistency is more important to me than the choice we make.

Maybe we need some official doc on this?